lib/commands/reshim.bash

```diff
@@ -97,7 +105,7 @@ write_shim_script() {
     cat <<EOF >"$shim_path"
 #!/usr/bin/env bash
 # asdf-plugin: ${plugin_name} ${version}
-exec $(asdf_dir)/bin/asdf exec "${executable_name}" "\$@" # asdf_allow: ' asdf '
```

Comment:
The function `asdf_dir` (https://github.com/asdf-vm/asdf/blob/master/lib/utils.bash#L22) seemingly checks for the value of `ASDF_DIR`, sets it if it doesn't exist, and then returns that same value. Are we hiding a bug in that functions behaviour by replacing it? Does anyone have time to investigate?

Reply:
Awaiting feedback

Reply:
> The function `asdf_dir` (https://github.com/asdf-vm/asdf/blob/master/lib/utils.bash#L22) seemingly checks for the value of `ASDF_DIR`, sets it if it doesn't exist, and then returns that same value. Are we hiding a bug in that functions behaviour by replacing it? Does anyone have time to investigate?

Good question. I believe the intent here is to replace `$(asdf_dir)`, which expanded to a path at **shim generation time** with syntax for the `$ASDF_DIR` variable, which is expanded at **shim execution time**, and therefore dynamically changes as `$ASDF_DIR` is changed, making shims more adaptable.

I need to spend some time thinking about this. It might be the right solution, or there may be a better solution. In either case I think we need some better testing around this.

Reply:
I wonder if instead of

``` bash
exec $(asdf_dir)/bin/asdf exec "${executable_name}" "\$@" # asdf_allow: ' asdf '
```

if we couldn't just do:

``` bash
exec asdf exec "${executable_name}" "\$@" # asdf_allow: ' asdf '
```

Should `asdf` already be on the `$PATH`? If so, why do we need to specify the absolute path of it? I need to investigate this.